### PR TITLE
Only unlock telemetry data that was locked before

### PR DESCRIPTION
**Pull request by schugabe**
This PR fixes a bug that prevents sending of telemetry data if only one type of message is received from the FC.

Unlocking the old tlm message did happen in the same method that did fetch the next tlm message to be sent.
If no other type of tlm message is received the unlock step did always clear the update flag and prevented any further communication.

To fix this bug the unlock is only done if the value was locked before.

After that a new message can be stored. The next call to GetNextPayload returns this message.

This issue would happen if betaflight is configured to send only the battery sensor.

### CODE REVIEW

File: src/lib/Telemetry/telemetry.cpp
Comment (CapnBry) [inline src/lib/Telemetry/telemetry.cpp:75]: Can't we get rid of `GetCurrentPayload()`? Nothing else uses it and `UnlockCurrentPayload` already checks the lock. It doesn't do anything and just adds confusion. Just Unlock and move on in rx_main and eliminate GetCurrentPayload altogether. The pseudocode looks better I think
```
if (!Active)
    Unlock
    if (GetNext(&x))
        Sender.SetNext(x)
```
EDIT oops well I should have clicked a line at the end of the function so you could see it here, sorry about that.

**Review by CapnBry — CHANGES_REQUESTED** [review]
For everyone else's benefit, I'm the person that brought this bug up. To test it without the patch, just go to the betaflight CLI and disable all the sensors you can:
```
# Disable Attitude telemetry item
set telemetry_disabled_pitch = ON
set telemetry_disabled_roll = ON
set telemetry_disabled_heading = ON
# Disable Battery telemetry item
set telemetry_disabled_voltage = ON
set telemetry_disabled_current = ON
set telemetry_disabled_fuel = ON
# Disable GPS telemetry item
set telemetry_disabled_altitude = ON
set telemetry_disabled_lat_long = ON
set telemetry_disabled_ground_speed = ON
set telemetry_disabled_heading = ON
```
This will just leave the mode, which can't be disabled currently but I have a PR #10745 in to Betaflight to be able to turn it off in CRSF. After the first update to `FM`, it will stop updating and drop off. The Sender currently lets the FC add update telemetry, but when Telemetry class looks to see what to send, it was resetting the flags on the updated telemetry item because it though it just sent that one, but it was new!

This looks like it will fix it, and testing with just battery stats I see just them updating frequently. I think `GetCurrentPayload` should be removed before merging.

**Comment by schugabe** [comment]
I came up with a simpler solution: only reset lock+update flag if it entry was locked before. That does not change the usage of the class and fixes the issue. I also removed the unused method.

**Comment by CapnBry** [comment]
Oh s**t yeah I like this a lot better, much easier to follow. I'll test this later today and approve if it passes testing. Looks like it will though.

**Review by CapnBry — APPROVED** [review]

**Comment by CapnBry** [comment]
Yeah this is great, really nice refactor. It is easier to read than the first version and addresses the issue cleanly. Tested with Betaflight with 1x and 2x items, and my CRServoF servo outputter with just its one item. We'll see who will merge this in my sleep tonight for RC6.

**Comment by AlessandroAU** [comment]
me! :)

**Pull request merged**
